pkg/smartcontract/manifest/manifest.go

```diff
@@ -135,3 +137,176 @@ func (m *Manifest) DecodeBinary(r *io.BinReader) {
+	}
+	return stackitem.NewStruct([]stackitem.Item{
+		stackitem.Make(m.Name),
+		stackitem.Make(groups),
+		stackitem.Make(supportedStandards),
+		abi,
+		stackitem.Make(permissions),
+		trusts,
+		extra,
+	}), nil
+}
+
+// FromStackItem converts stackitem.Item to Manifest.
+func (m *Manifest) FromStackItem(item stackitem.Item) error {
+	str, ok := item.Value().([]stackitem.Item)
+	if !ok || len(str) != 7 {
+		return errors.New("invalid stackitem type or length")
+	}
+	name, err := str[0].TryBytes()
+	if err != nil {
+		return err
+	}
+	m.Name = string(name)
+	groups, ok := str[1].Value().([]stackitem.Item)
+	if !ok {
+		return errors.New("invalid Groups stackitem type")
+	}
+	m.Groups = make([]Group, len(groups))
+	for i := range groups {
+		group := new(Group)
+		err := group.FromStackItem(groups[i])
+		if err != nil {
+			return err
+		}
+		m.Groups[i] = *group
+	}
+	supportedStandards, ok := str[2].Value().([]stackitem.Item)
+	if !ok {
+		return errors.New("invalid SupportedStandards stackitem type")
+	}
+	m.SupportedStandards = make([]string, len(supportedStandards))
+	for i := range supportedStandards {
+		s, err := supportedStandards[i].TryBytes()
+		if err != nil {
+			return err
+		}
+		m.SupportedStandards[i] = string(s)
+	}
+	abi := new(ABI)
+	if err := abi.FromStackItem(str[3]); err != nil {
+		return err
+	}
+	m.ABI = *abi
+	permissions, ok := str[4].Value().([]stackitem.Item)
+	if !ok {
+		return errors.New("invalid Permissions stackitem type")
+	}
+	m.Permissions = make([]Permission, len(permissions))
+	for i := range permissions {
+		p := new(Permission)
+		if err := p.FromStackItem(permissions[i]); err != nil {
+			return err
+		}
+		m.Permissions[i] = *p
+	}
+	if _, ok := str[5].(stackitem.Null); ok {
+		m.Trusts.Restrict()
+	} else {
+		trusts, ok := str[5].Value().([]stackitem.Item)
+		if !ok {
+			return errors.New("invalid Trusts stackitem type")
+		}
+		m.Trusts = WildUint160s{Value: make([]util.Uint160, len(trusts))}
+		for i := range trusts {
+			bytes, err := trusts[i].TryBytes()
+			if err != nil {
+				return err
+			}
+			m.Trusts.Value[i], err = util.Uint160DecodeBytesBE(bytes)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	extra, err := str[6].TryBytes()
+	if err != nil {
+		return err
+	}
+	if string(extra) == "null" {
+		return nil
+	}
+	return json.Unmarshal(extra, &m.Extra)
+}
+
+// ToStackItem converts ABI to stackitem.Item.
+func (a *ABI) ToStackItem() stackitem.Item {
+	methods := make([]stackitem.Item, len(a.Methods))
+	for i := range a.Methods {
+		methods[i] = a.Methods[i].ToStackItem()
+	}
+	events := make([]stackitem.Item, len(a.Events))
+	for i := range a.Events {
+		events[i] = a.Events[i].ToStackItem()
+	}
+	return stackitem.NewStruct([]stackitem.Item{
+		stackitem.Make(methods),
+		stackitem.Make(events),
+	})
+}
+
+// FromStackItem converts stackitem.Item to ABI.
+func (a *ABI) FromStackItem(item stackitem.Item) error {
+	str, ok := item.Value().([]stackitem.Item)
```

Comment:
It can be an array, but C# node will fail to unwrap an array, it only works with structs.